packages/app/src/runner/useRunnerStyle.ts

```diff
@@ -67,10 +53,18 @@ export const useRunnerStyle = ({
   })
 
   const viewportStyle = computed(() => {
-    return `
-      width: ${autStore.viewportDimensions.width}px;
-      height: ${autStore.viewportDimensions.height}px;
-      transform: scale(${scale.value});`
+    let style = `
+    width: ${autStore.viewportDimensions.width}px;
+    height: ${autStore.viewportDimensions.height}px;
+    transform: scale(${scale.value});
+    `
+
+    if (!screenshotStore.isScreenshotting) {
+      style += `
+      margin-left: ${(containerWidth.value / 2) - (autStore.viewportDimensions.width / 2) }px`
```

Comment:
How did you arrive at this logic? I wonder if this is worth commenting - it reads like black magic, haha. This is for centering the AUT, correct?

I'm really surprised we can center using `margin` at all - wouldn't this usually just be something like `flex justify-center` on the parent element? I'm guessing there's a good reason for this, but I don't know what it is (thus a comment might be good).

Reply:
I can put a comment. Short version is: I arrived at it by looking at `develop`, we already do it there.

There was a centering issue @ZachJW34 pointed out, particularly obvious with portrait viewports. When we have a perfectly centered element (with traditional means) and do a CSS transform to scale it, it the transform doesn't change the actual space the element takes up in the document flow, so margin hacks like this end up being pretty common. I wasn't surprised when I went looking to see how we solved it in 9.x and we were programmatically resetting margin with JS.